### DISCUSSION TITLE
Trying to replicate the example in the Mainline Development documentation and versioning is not incrementing as expected; Am I missing something here?

### DISCUSSION THREAD

**Discussion by davidandersen** [GitTools/GitVersion#2901] [Q&A] [answered]
I thought I found some issues with `GitVersion` when using on my local and within GitHub actions on another project, but I took a step back and thought that I would attempt to replicate the example of what **Mainline Development** looks like from the [documentation](https://gitversion.net/docs/reference/modes/mainline).

I was able to match to the documented versions up through `1.1.0+4`, but when I created the branch `feature/foo3`, made a commit, and merged with `main` using `+semver: minor` in the merge message, the version did not increment to `1.2.0+6`. Instead, it stayed as `1.1.0`.

I proceeded to repeat the process, but using `+semver: minor` in the commit message, then merging, and the version still was not incremented to `1.2.0`.

Does `Mainline` require me to tag `main` in order for versions to be incremented using `+semver: [ type ]`? I was under the impression that `GitVersion` will increment regardless of whether or not there is an explicit tag (but I am aware of the best practice to tag for performance reasons).

Click [here](https://github.com/davidandersen/gitversion-mainline) for a link to the repo that I created that outlines the steps that I took to generate the versions.

If you go to the **Actions** tab, you will be able to see the history of the builds. Click [here](https://github.com/davidandersen/gitversion-mainline/actions/runs/1428301275) to be taken to the merge for `feature/foo3`.

In the [build script](https://github.com/davidandersen/gitversion-mainline/blob/main/.github/workflows/version.yml) there is a section where version is printed to the screen with a name of `Run echo 1.1.0`.

The configuration file being used can be found [here](https://github.com/davidandersen/gitversion-mainline/blob/main/GitVersion.yml).

Here is a screen shot of the version in the diagram from the documentation that illustrates where I'm failing to produce the correct version.

![image](https://user-images.githubusercontent.com/2736161/140598582-528fcbd4-1253-47cd-9763-7175fdfb00aa.png)

Any feedback would be greatly appreciated! Thanks!

**Comment by asbjornu** [comment, marked as answer]
It seems like you've configured all of your branches to use `mode: ContinuousDelivery`. To understand how Mainline works in GitVersion, I would recommend that you reduce your `GitVersion.yml` file to nothing but:

```yaml
mode: Mainline
```

You may also consider fast-forward merging squashed commits of your feature branches into `main` instead of adding merge-commits. That is closer to how [GitHub Flow](https://docs.github.com/en/get-started/quickstart/github-flow) (which is a type of Mainline flow) works.
